fuel-core/src/executor.rs

```diff
@@ -1169,6 +1169,9 @@ impl Executor {
                 TransactionStatus::Failed { block_id, .. } => {
                     *block_id = finalized_block_id;
                 }
+                TransactionStatus::SqueezedOut { .. } => {
```

Comment:
This could introduce a new race condition for the `DuplicateTransactionID` check in the executor, if the status isn't updated synchronously with execution.

Reply:
<img width="708" alt="image" src="https://user-images.githubusercontent.com/18346821/200717830-7988f955-46ea-48bf-868d-9fab31db6c57.png">

We use existence in the `Transactions`, not statuses=)

Reply:
ah ok, well in that case we could just add the tx statuses to the execution result and let the txpool update them. I think we should do this as a followup PR though.